src/test/java/org/rxbooter/injector/core/KeyTest.java

```diff
@@ -78,6 +78,7 @@ public void keyFollowsEqualsContract() throws Exception {
         assertThat(key1).isEqualTo(key1);
         assertThat(key2).isEqualTo(key2);
 
+        //Key compared to string ??
         assertThat(key1.equals("")).isFalse();
```

Comment:
This test verifies that .equals() contract is followed. And .equals() can receive any other object as input, so check is perfectly legal and validates that objects of other type will be considered not equal.